### PR TITLE
Migrate to ESLint

### LINKED ISSUE #40: Migrate to ESLint
Since yesterday [yeoman/generator-node](https://github.com/yeoman/generator-node) use and provide ESLint with shareable config  [eslint-config-xo-space](https://github.com/sindresorhus/eslint-config-xo-space).

PR accepted ?

### PR DESCRIPTION
**Pull request by zckrs**
fix #40

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -15,7 +15,7 @@ var keyFilename = process.env.GCLOUD_KEYFILENAME;
 var log = process.env.LOGGER || console;
 
 var gcloud = require('gcloud');
-var pluginCache = require('./plugin-cache');
+var PluginCache = require('./plugin-cache');
```
Comment (hemanth) [inline index.js]: :+1:

File: package.json [hunk 2]
```diff
@@ -32,8 +32,19 @@
     "q": "^1.4.1"
   },
   "devDependencies": {
+    "eslint": "^1.10.3",
     "mocha": "^2.2.5"
   },
+  "eslintConfig": {
+    "extends": "xo-space",
+    "rules": {
+      "camelcase": 0,
```
Comment (sindresorhus) [inline package.json]: Why did you have to disable this?

File: package.json [hunk 3]
```diff
@@ -32,8 +32,19 @@
...
+      "no-inline-comments": 0
```
Comment (sindresorhus) [inline package.json]: Why this?

**Comment by hemanth** [comment]
@zckrs Reverted it.

File: package.json (hunk 2 above)
Comment (zckrs) [inline package.json]:
```
/home/zckrs/github/yeoman-generator-list/newrelic.js
  11:3  error  Identifier 'app_name' is not in camel case     camelcase
  15:3  error  Identifier 'license_key' is not in camel case  camelcase

/home/zckrs/github/yeoman-generator-list/plugin-cache/src/gh-stats.js
  41:11  error  Identifier 'client_id' is not in camel case      camelcase
  42:11  error  Identifier 'client_secret' is not in camel case  camelcase

```

I'm not sure I can fix error without breaking application

Comment (zckrs) [inline package.json]: Can I disable rule for these lines  ?

File: package.json (hunk 3 above)
Comment (zckrs) [inline package.json]:
```
/home/zckrs/github/yeoman-generator-list/plugin-cache/src/merge-info.js
  9:27  error  Unexpected comment inline with code  no-inline-comments
```
Can I move this comment ?

File: package.json (hunk 2 above)
Comment (sindresorhus) [inline package.json]: For `/home/zckrs/github/yeoman-generator-list/newrelic.js` just disable the rule in the file with `/* eslint-disable camelcase */` at the top.

With `/home/zckrs/github/yeoman-generator-list/plugin-cache/src/gh-stats.js` do the same, but just inline.

File: package.json (hunk 3 above)
Comment (sindresorhus) [inline package.json]: Sure

**Comment by zckrs** [comment]
Done ===>  https://github.com/yeoman/yeoman-generator-list/pull/44

**Pull request merged**